sentry/src/routes/channel.rs

```diff
@@ -342,6 +336,74 @@ pub async fn get_spender_limits<A: Adapter + 'static>(
     Ok(success_response(serde_json::to_string(&res)?))
 }
 
+pub async fn get_all_spender_limits<A: Adapter + 'static>(
+    req: Request<Body>,
+    app: &Application<A>,
+) -> Result<Response<Body>, ResponseError> {
+    let channel = req
+        .extensions()
+        .get::<ChannelV5>()
+        .expect("Request should have Channel")
+        .to_owned();
+
+    let new_state = match get_corresponding_new_state(&app.pool, &channel).await? {
+        Some(new_state) => new_state,
+        None => return Err(ResponseError::NotFound),
+    };
+
+    let mut all_spender_limits: HashMap<Address, Spender> = HashMap::new();
+
+    let all_spendables = get_all_spendables_for_channel(app.pool.clone(), &channel.id()).await?;
+
+    // Using for loop to avoid async closures
+    for spendable in all_spendables {
+        let spender = spendable.spender;
+        let spender_leaf = if new_state.msg.balances.spenders.contains_key(&spender) {
+            let balance = new_state.msg.balances.spenders.get(&spender).unwrap();
+
+            Some(SpenderLeaf {
+                total_spent: spendable
+                    .deposit
+                    .total
+                    .checked_sub(balance)
+                    .unwrap_or_default(),
+                // merkle_proof: [u8; 32], // TODO
+            })
+        } else {
+            None
+        };
```

Comment:
For this case we have `Option`, you don't need to go over the trouble for checking `contains_key()` & then `unwrap()`

```suggestion
        let spender_leaf = new_state.msg.balances.spenders.get(&spender).map(|balance| {
             SpenderLeaf {
                total_spent: spendable
                    .deposit
                    .total
                    .checked_sub(balance)
                    .unwrap_or_default(),
                // merkle_proof: [u8; 32], // TODO
            }
        });
```